### DISCUSSION TITLE
How can I configure IIS settings in ASP.NET Core?

### DISCUSSION THREAD

**Discussion by SetTrend** [dotnet/aspnetcore#57612] [Q&A] [answered]
### Description

The MS Learn documentation page "[In-process hosting with IIS and ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/in-process-hosting)" appears to provide incorrect information in section "[Application configuration](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/in-process-hosting#application-configuration)":

```c#
builder.Services.Configure<IISServerOptions>(options =>
{
  options.AutomaticAuthentication = false;
});
```

The `Services` object available to me doesn't provide a `Configure` method:

![No Services Configure](https://github.com/user-attachments/assets/980376d7-1517-4e87-a8d2-decc34b1863b)

How can I configure IIS module settings in ASP.NET Core 8<sup>++</sup>?

**Comment by martincostello** [comment, marked as answer]
Do you have the `Microsoft.Extensions.Options` (or it might be `Microsoft.Extensions.DependencyInjection`) namespace in scope in your `using`s?

**Reply by SetTrend** [reply]
Ah, I see. Thanks for helping me out on this!

So, the documentation is missing that using (`Microsoft.Extensions.DependencyInjection`).

**Reply by martincostello** [reply]
Are you using `<ImplicitUsings>true</ImplicitUsings>` in your project file? If not that will be why you had to add it.

The documentation samples appear to be based around minimal hosting for .NET 6+, where this is the default, so they don't need explicitly adding as they're implicitly added to every code file of a project using Microsoft.NET.Sdk.Web.

**Reply by SetTrend** [reply]
Ah, I see. I'm a "control freak". So, I'd like to know where things are coming from. And that's why I disabled this setting 😉
